### PR TITLE
feat: built a static interface for the IWD

### PR DESCRIPTION
**Pull request by Adejuwon-00**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+let names=[]
+function myFunction()
```
Comment (gwa-admin) [inline index.js:2]: Please use descriptive variable and function names.

Comment (gwa-admin) [reply index.js:2]: Good function name, but going forward can we use [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) except where unavoidable?

So something like `const addNamesToList = () => {} `

File: index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+{
+ let x=document.getElementById("box");
```
Comment (gwa-admin) [inline index.js:4]: Is there we are using `let` instead of `const`?

File: index.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  names.push(document.getElementById("name").value);
+ x.innerHTML=names.join('<br/><br/>'); 
+}
+
+//for accessibility
+let input = document.getElementById("name");
```
Comment (gwa-admin) [inline index.js:10]: Since this variable isn't changing, we should use `const` instead of `let`. Please refer to this [article](https://www.freecodecamp.org/news/var-let-and-const-whats-the-difference/) to understand the difference

File: styles.css [hunk 4]
```diff
@@ -0,0 +1,101 @@
+@import url('https://fonts.googleapis.com/css2?family=Archivo:wght@300;400;600&display=swap');
+
+* {
+    --font-family: 'Archivo', sans-serif;
```
Comment (gwa-admin) [inline styles.css:4]: Good use of CSS variables. Great job!

File: index.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+input.addEventListener("keypress", function(event) {
+  if (event.key === "Enter") {
```
Comment (gwa-admin) [inline index.js:12]: Really good job with making this functionality accessible. Very commendable! 👍

File: index.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+    event.preventDefault();
+    document.getElementById("add-btn").click();
```
Comment (gwa-admin) [inline index.js:14]: What does this last line of code do on line 14 do?

File: styles.css [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+h1{
+    line-height: .8;
+    /* for media queries */
+    font-size: clamp(2em, 3vw, 2.5rem);
+}
+
+h3{
+    line-height: .5;
+    font-size: clamp(1.05em, 1.5vw, 1.5rem);
```
Comment (gwa-admin) [inline styles.css:43]: What does `clamp` do?

File: styles.css [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+input{
+  padding:10px;
+  width:auto;
+  background-color: var(--input-bg);
+  max-width:325px;
+  height:auto;
+  border:none;
+  text-align:center;
+  -webkit-transition:0.2s ease all;
```
Comment (gwa-admin) [inline styles.css:77]: What does this do? Why did you add it?

File: styles.css [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+::placeholder {
+    text-align: left;
+    color: black;
+}
+
+#box{
+    border:.5px solid black;
+    border-radius: 25px;
+    width:28.78em;
```
Comment (gwa-admin) [inline styles.css:92]: As much as is possible, I'd recommend `rem` over `em`.

File: index.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+let names=[]
+function myFunction()
+{
+ let x=document.getElementById("box");
+  names.push(document.getElementById("name").value);
+ x.innerHTML=names.join('<br/><br/>');
```
Comment (gwa-admin) [inline index.js:6]: Your logic doesn't **add the names as a list of names**.

The expectation is that each name that is added should be an item in a list so that it is rendered as a list, and is accessible as shown below:

```
<ol>
  <li>Name 1</li>
  <li>Name 2</li>
<ol>

```

**Review by gwa-admin — COMMENTED** [review]
Good job 👍 on this task @Adejuwon-00, I've added some comments for you on the PR. Here's a summary:

**What you did well.**
- The user interface of the app meets design specifications.
- Showed good knowledge of HTML and CSS, including the use of CSS variables.
- Good thinking about accessibility when you added logic that would allow the form work when the user presses the enter key.

**Things you can do better on**
- Use more descriptive function names.
- I want to see you start using CSS variables for some of your values.
- Endeavor to write semantic HTML code, like in the case where you could have used an ordered list instead of just a paragraph.
- Know when to `const` instead of `let`.
- Ensure that you write robust code that checks for edge cases. Your JavaScript logic doesn't account for the edge case where the input is empty and the user clicks on the **ADD** button, which is why you see what is shown below.


image

File: index.js (hunk 1 above)
Comment (Adejuwon-00) [reply index.js:2]: Alright, sir..I'll refactor the code and submit a PR

File: index.js (hunk 3 above)
Comment (Adejuwon-00) [reply index.js:10]: Alright, sir.

File: index.js (hunk 2 above)
Comment (Adejuwon-00) [reply index.js:4]: I understand it's a variable with a fixed value.....I'll refactor the code and create a PR

File: styles.css (hunk 4 above)
Comment (Adejuwon-00) [reply styles.css:4]: Haha!.....Thank you

File: index.js (hunk 6 above)
Comment (Adejuwon-00) [reply index.js:14]: It'll allow for the trigger of the HTML element with the ID of "add-btn" with the click event handler to act as if it was actually clicked ( say it's a click event simulator). This enables the user to add and display the data from the input by tapping the enter button

File: styles.css (hunk 7 above)
Comment (Adejuwon-00) [reply styles.css:43]: It's a CSS method used for responsiveness in general. It takes in 3 parameters (the first in this case is the minimum font size, the second parameter is the preferred font size and the last is the maximum font size). Attached is an article that explains more about [clamp](https://developer.mozilla.org/en-US/docs/Web/CSS/clampl)

File: styles.css (hunk 8 above)
Comment (Adejuwon-00) [reply styles.css:77]: This was a mistake!

File: styles.css (hunk 9 above)
Comment (Adejuwon-00) [reply styles.css:92]: Very true and I'll do well to fix it. I actually just read this [article](https://refine.dev/blog/rem-vs-em/) for best practices.

File: index.js (hunk 10 above)
Comment (Adejuwon-00) [reply index.js:6]: Alright, Sir.
Thank you for your review and feedback on my PR. Your attention to detail and keen eye for errors really helped me catch mistakes that I missed, and I'm grateful for your input. I'll do well to refactor and submit a PR

**Comment by gwa-admin** [comment]
Good job @Adejuwon-00 with refactoring and making the changes.
Closing this PR now to cleanup the repo for our project, FYI.
